File: python/6_kyu/ten_minute_walk.py
# https://www.codewars.com/kata/54da539698b8a2ad76000228

def is_valid_walk(walk):
    directions, x, y = {"n": 1, "s": -1, "e": 1, "w": -1}, 0, 0
    if len(walk) == 10:
        for axis in walk:
            if (axis == "w" or axis == "e"):
                x += directions[axis]
            else:
                y += directions[axis]
        return True if x == 0 and y == 0 else False
    else:
        return False

File: CodeWars/javascript/6_kyu/ten_minute_walk.js
// https://www.codewars.com/kata/54da539698b8a2ad76000228

function isValidWalk(walk) {
  if (walk.length !== 10) return false;
  const x_axis = ["w", "e"];
  let x = 0;
  let y = 0;
  for (d of walk) {
    if (x_axis.includes(d)) x += d === "e" ? 1 : -1;
    else y += d === "n" ? 1 : -1;
  }
  return x === 0 && y === 0;
}
